src/common/threadpool.cpp

```diff
@@ -73,12 +72,12 @@ void threadpool::submit(waiter *obj, std::function<void()> f) {
   } else {
     if (obj)
       obj->inc();
-    queue.push_back(e);
+    queue.push_back({obj, f});
```

Comment:
Was this change necessary? The original code is allocating early outside the lock to minimize time in the lock.

Reply:
No, this was to avoid a temp object, I will revert.